.github/actions/kibana-docker-image/action.yml

```diff
@@ -0,0 +1,87 @@
+name: Kibana Docker Image
+description: |
+  An Action to build and push Kibana docker images given a git ref.
+inputs:
+  git-repository:
+    description: The git repository to build the image from.
+    default: "elastic/kibana"
+    required: false
+  git-ref:
+    description: "The git ref of elastic/kibana to build the image from. (Default: default branch)"
+    required: false
+  serverless:
+    description: Whether to build the serverless image or not.
+    required: false
+    default: "false"
+  docker-registry:
+    description: Docker registry to publish the image to.
+    required: false
+    default: "docker.elastic.co"
+  docker-namespace:
+    description: Docker namespace to publish the image to.
+    required: false
+    default: "observability-ci"
+
+outputs:
+  ref:
+    description: The reference of the Docker image that was built in the format image:tag.
+    value: ${{ steps.vars.outputs.docker-reference }}
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v4
+      with:
+        repository: ${{ inputs.git-repository }}
+        ref: ${{ inputs.git-ref }}
+        path: kibana-repo-checkout
+
+    - uses: actions/setup-node@v3
+      with:
+        node-version-file: kibana-repo-checkout/.node-version
+        cache: yarn
+        cache-dependency-path: kibana-repo-checkout/yarn.lock
+
+    - name: Set up variables
+      id: vars
+      working-directory: kibana-repo-checkout
+      run: |
```

Comment:
**nit**: what do you think moving this part to a script?

Reply:
https://github.com/elastic/apm-pipeline-library/pull/2348/commits/f1a719deb787b8593e57e8c456a2b03783bc1a86